docs/docs/guides/evaluation/scorers.md

```diff
@@ -0,0 +1,343 @@
+# Evaluation Metrics
+
+## Evaluations in Weave
+In Weave, Scorers are used to evaluate AI outputs and return evaluation metrics. They take the AI's output, analyze it, and return a dictionary of results. Scorers can use your input data as reference if needed and can also output extra information, such as explanations or reasonings from the evaluation.
+
+Scorers are passed to a `weave.Evaluation` object during evaluation. There are two types of Scorers in weave:
+
+1. **Function-based Scorers:** Simple Python functions decorated with `@weave.op`.
+2. **Class-based Scorers:** Python classes that inherit from `weave.Scorer` for more complex evaluations.
+
+Scorers must return a dictionary and can return multiple metrics, nested metrics and non-numeric values such as text returned from a LLM-evaluator about its reasoning.
+
+## Function-based Scorers
+These are functions decorated with `@weave.op` that return a dictionary. They're great for simple evaluations like:
+
+```python
+@weave.op
+def evaluate_uppercase(text: str):
+    return {"text_is_uppercase": text.isupper()}
+
+eval = weave.Evaluations(..., scorers=[evaluate_uppercase])
+```
+
+When the evaluation is run, `evaluate_uppercase` checks if the text is all uppercase.
+
+## Class-based Scorers
+For more advanced evaluations, especially when you need to keep track of additional scorer metadata, try different prompts for your LLM-evaluators, or make multiple function calls, you can use the `Scorer` class.
+
+**Requirements:**
+1. Inherit from `weave.Scorer`.
+2. Define a `score` method decorated with `@weave.op`.
+3. The `score` method must return a dictionary.
+
+Example:
+
+
+```python
+from weave import Scorer
+from weave.scorers import create
+
+llm_client = ...
+
+class SummarizationScorer(Scorer):
+    model_id: str = "the LLM model to use"
+    system_prompt: str = "Evaluate whether the summary is good."
+
+    @weave.op
```

Comment:
I notice that you seem to decorate all of the functions inside Scorers.  Do you think this should be done automatically for any non-underscore-prefixed func?  Or would the lack of decoration make it less obvious?

Reply:
No I think whether or not it gets decorated depends on what the function does